File: surf-cloud-core/src/main/java/dev/slne/surf/cloud/core/netty/ServerState.java
package dev.slne.surf.cloud.core.netty;

public enum ServerState {
  RESTARTING(false),
  LOBBY(true),
  OFFLINE(false),
  ONLINE(true);

  private final boolean allowJoin;

  private ServerState(boolean allowJoin) {
    this.allowJoin = allowJoin;
  }

  public boolean allowJoin() {
    return allowJoin;
  }
}

File: surf-cloud-api/surf-cloud-api-common/src/main/kotlin/dev/slne/surf/cloud/api/common/server/state/ServerState.kt
package dev.slne.surf.cloud.api.common.server.state

import dev.slne.surf.cloud.api.common.netty.network.codec.ByteBufCodecs
import dev.slne.surf.cloud.api.common.util.ByIdMap

enum class ServerState(val id: Int, val allowJoin: Boolean) {
    RESTARTING(0, false),
    LOBBY(1, true),
    ONLINE(2, true),
    OFFLINE(3, false);

    companion object {
        val BY_ID = ByIdMap.continuous(
            ServerState::id,
            entries.toTypedArray(),
            ByIdMap.OutOfBoundsStrategy.LAST
        )

        val STREAM_CODEC = ByteBufCodecs.idMapper(BY_ID, ServerState::id)
    }
}
